JS/imdb.js

```diff
@@ -0,0 +1,143 @@
+
+function addMovie (e) {
+
+  //assign to movie@ the relevant text element
+  var movieName = document.getElementById("movieName");
+  var movieJanner = document.getElementById("movieJanner");
+  var movieTime = document.getElementById("movieTime");
+  
+  //create a button element that will be used to delete list item
+  var delButton = document.createElement("button");
+  delButton.setAttribute('value', 'X');
+  //delButton.setAttribute('onclick',deleteLi());
+  var buttonData = document.createTextNode("X");
+  delButton.appendChild(buttonData);
+
+
+  //create new "li" item in order to insert it to the existing "ul"
+  var newItem = document.createElement("li");
+
+	 
+  //"create" text node to assign in the new "li" 
+  var newName = document.createTextNode(movieName.value + ", ");
+  //assinging the text node to the new "li"
+  newItem.appendChild(newName);
+
+  //"create" another text node to append in the new "li" 
+  var newJanner = document.createTextNode(movieJanner.value + ", ");
+  //appending the text node to the new "li"
+  newItem.appendChild(newJanner);
+
+  //"create" another text node to append in the new "li" 
+  var newTime = document.createTextNode(movieTime.value + " ");
+  //appending the text node to the new "li"
+  newItem.appendChild(newTime);
+  //appending the the delete nutton to the "li" element
+  newItem.appendChild(delButton);
+  
+  //push the "li" to the stse array
+  liArray.push(newItem);
+  
+  //call showList with the text in the null as a value
+  showList(null);
+ 
+  
+};
+
+
+//this function purpose is to prevent sumbitting 
+//unfull movie data.
+function disCheck() {
+	
+	// assign to movie@ the relevant text element
+	var movieName = document.getElementById("movieName");
+	var movieJanner = document.getElementById("movieJanner");
+	var movieTime = document.getElementById("movieTime");
+	
+	//condition for checking false insertion of "li"'s
+	if ((movieName.value !== "") && (movieJanner.value !== "") && 
+		(movieTime.value !== "")) {
+
+		document.getElementById("buttn").disabled = false;
+	
+	}else {
+		document.getElementById("buttn").disabled = true;
+	}
+}
+
+
+//this function is responsible for presenting the item in the ul.
+function showList (e) {
+
+  //// temporary - checking myself
+  console.log(e);
+
+  //in case e == null (probably when calling the function 
+  //from addMovie or when serach box is empty). 
+  //will present all the elements in the array- all the list.
+  if (e == null){
+ 
+   for (var i = 0; i < liArray.length; i++) {
+
+   	 var currentLi = document.getElementById("anchor"); 
+  	 var ul = document.getElementById("mList");
+  	 ul.insertBefore(liArray[i], currentLi);
+   
+   } 
+  //in case e !== null (probably when calling the function 
+  //from serach box and the serch box is not empty). 
+  //will present all the elements in the array that 
+  //the string in the serch bow is a substring of them.
+  }else {
+     
+     for (var i = 0; i < liArray.length; i++) {
```

Comment:
@oregilron I liked this function and the using of the array, but let's make it even better, instead of this for loop use the Array.prototype method.
I want you to replace this for loop with two Array.prototype methods. i won't tell you which because I want you to go over all the methods and then decide what to use